GitCommands/Git/GitModule.cs

```diff
@@ -386,7 +386,7 @@ public string GitCommonDirectory
                 {
                     var commDir = RunGitCmdResult("rev-parse --git-common-dir");
                     _gitCommonDirectory = PathUtil.ToNativePath(commDir.StdOutput.Trim());
-                    if (!commDir.ExitedSuccessfully || _gitCommonDirectory == ".git" || !Directory.Exists(_gitCommonDirectory))
+                    if (!commDir.ExitedSuccessfully || _gitCommonDirectory == ".git" || _gitCommonDirectory == "." || !Directory.Exists(_gitCommonDirectory))
```

Comment:
This was a strange one. `git rev-parse --git-common-dir` was returning `"."` in the case of `git rebase -i`, but returning `".git"` in the case of `git commit`. I can't see why we wouldn't add `"."` to the list of paths exchanged for the full git directory.

Reply:
please submit this commit as a separate PR

Reply:
Done in #4871.